fdbserver/include/fdbserver/DDTeamCollection.h

```diff
@@ -240,6 +240,9 @@ class DDTeamCollection : public ReferenceCounted<DDTeamCollection> {
 
 	std::vector<Reference<TCTeamInfo>> badTeams;
 	std::vector<Reference<TCTeamInfo>> largeTeams;
+	// shards have been over-replicated
+	std::map<int, int> largeTeamSizeShards;
```

Comment:
nit: do we need map to be sorted, can we use `std::unordered_map` for better lookup performance?

Reply:
I think the iteration performance might be more critical. Though I think the map won't be large so not a big deal...

Reply:
Thanks, but, I think my basic question remains unanswered, do we need the map to be sorted. If not, then, not sure what's the gain of using a sorted map, as both insertion and lookup are costly for sorted map (also deletion incase we do it)

Reply:
Because when iterator over the map I hope the over-replication size sorts the output.